Sources_v3/Controllers/CurrentUserController_Tests.swift

```diff
@@ -0,0 +1,355 @@
+
+        // Start updating
+        let startUpdatingError = try await(controller.startUpdating)
+        
+        // Assert `startUpdating` finished without any error
+        XCTAssertNil(startUpdatingError)
+
+        // Simulate saving current user to a database
+        let createUserError = try await {
+            client.databaseContainer.write({ session in
+                try session.saveCurrentUser(payload: currentUserPayload)
+            }, completion: $0)
+        }
+        
+        // Assert saving current user to a database finished without any error
+        XCTAssertNil(createUserError)
+        
+        // Update current user data
+        extraData = NameAndImageExtraData(name: .unique, imageURL: .unique())
+        currentUserPayload = .dummy(userId: currentUserPayload.id,
+                                    role: currentUserPayload.role,
+                                    extraData: extraData)
+        
+        // Simulate updating current user in a database
+        let updateUserError = try await {
+            client.databaseContainer.write({ session in
+                try session.saveCurrentUser(payload: currentUserPayload)
+            }, completion: $0)
+        }
+        
+        // Assert updating current user to a database finished without any error
+        XCTAssertNil(updateUserError)
+        
+        // Assert delegate received correct entity change
+        AssertAsync {
+            Assert.willBeEqual(delegate.didChangeCurrentUser_change?.fieldChange(\.id), .update(currentUserPayload.id))
+            Assert.willBeEqual(delegate.didChangeCurrentUser_change?.fieldChange(\.extraData), .update(extraData))
+        }
+    }
+
+    func test_delegate_isNotifiedAboutUnreadCount_whenUserIsCreated() throws {
+        let unreadCount = UnreadCount(channels: 10, messages: 15)
+        
+        // Set the delegate
+        let delegate = TestDelegate()
+        delegate.expectedQueueId = controllerCallbackQueueID
+        controller.delegate = delegate
+
+        // Start updating
+        let startUpdatingError = try await(controller.startUpdating)
+        
+        // Assert `startUpdating` finished without any error
+        XCTAssertNil(startUpdatingError)
+
+        // Simulate saving current user to a database
+        let createUserError = try await {
+            client.databaseContainer.write({ session in
+                let currentUserPayload: CurrentUserPayload<DefaultDataTypes.User> = .dummy(userId: .unique,
+                                                                                           role: .user,
+                                                                                           unreadCount: unreadCount)
+                try session.saveCurrentUser(payload: currentUserPayload)
+            }, completion: $0)
+        }
+        
+        // Assert saving current user to a database finished without any error
+        XCTAssertNil(createUserError)
```

Comment:
☝️

Reply:
fixed in 9b3c74ee2d704845b8cd83448ade208720df8d94